src/animation/AnimationMixer.js

```diff
@@ -83,8 +83,10 @@ Object.assign( THREE.AnimationMixer.prototype, THREE.EventDispatcher.prototype,
 
 		var root = optionalRoot || this._root,
 			rootUuid = root.uuid,
-			clipName = ( typeof clip === 'string' ) ? clip : clip.name,
-			actionsForClip = this._actionsByClip[ clipName ];
+			clipObject = typeof clip === 'string' ? this.getClipByName( clip ) : clip,
```

Comment:
@sunag 
There's this issue that multiple objects can have different clips sharing the same name and it's actually not all too uncommon: 
Different characters that can all perform the same actions may e.g. "walk" differently an thus use different clips with that same name.

Instead of searching through existing actions, I guess it would be best to do something like `AnimationClip.findByName( root.animationClips, clip )`. Note that there's `AnimationClip.findByName`, but I'm not sure whether `root.animationClips` exists, or what its name is in case it does, or whether it would be the only possible name :). However, the nice thing about this approach would be that you could always pass a name to the mixer and it will always give you an action, currently playing or not, and create an action on demand, even if the clip has never been passed-in directly. I guess this may simplify higher level code quite a bit.

A less preferable fix would be to make it `this._getClipByNameAndRoot( clip, rootUuid )` and also compare against `action.getRoot().uuid` searching through the actions. This way the collision is resolved, but the limitation that the mixer must first see a clip "by object" before it can be addressed by name would remain. Also, the linear search through all cached actions is somewhat undesirable (there can be many of them).